src/sgame/sg_bot.cpp

```diff
@@ -503,11 +503,18 @@ void G_BotSpectatorThink( gentity_t *self )
 	self->botMind->futureAimTimeInterval = 0;
 	self->botMind->runningNodes.clear();
 
+	//FIXME: duplicate of sg_cmds.cpp:883 function "void Cmd_Team_f( gentity_t * )"
+	if ( g_doWarmup.Get() && ( ( level.warmupTime - level.time ) / 1000 ) > 0 )
```

Comment:
This could go a bit earlier, after the PMF_QUEUED if block. Otherwise LGTM

Reply:
By precaution, I'd rather keep the cleanup stage ran each botSpectatorThink. The stuff really feels hackish to me, I won't lie, but I do not understand enough yet the process of spawning bots.